### ISSUE TITLE
Could not find the End of Central Directory Record

### ISSUE THREAD

**Issue by Varunram** [nwjs-community/nw-builder#466]
Related to #132, but opening a new thread since there was no fix on that thread.

I get the following stack trace when I try to build some cross platform apps:
```
Error: Could not find the End of Central Directory Record
    at DecompressZip.findEndOfDirectory (/usr/local/lib/node_modules/nwjs-builder/node_modules/decompress-zip/lib/decompress-zip.js:165:15)
    at _fulfilled (/usr/local/lib/node_modules/nwjs-builder/node_modules/q/q.js:854:54)
    at self.promiseDispatch.done (/usr/local/lib/node_modules/nwjs-builder/node_modules/q/q.js:883:30)
    at Promise.promise.promiseDispatch (/usr/local/lib/node_modules/nwjs-builder/node_modules/q/q.js:816:13)
    at /usr/local/lib/node_modules/nwjs-builder/node_modules/q/q.js:624:44
    at runSingle (/usr/local/lib/node_modules/nwjs-builder/node_modules/q/q.js:137:13)
    at flush (/usr/local/lib/node_modules/nwjs-builder/node_modules/q/q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```
This occurs when I try to build for `win64`. When i build for other platforms, there are no errors, but the application doesn't work properly.

Any fixes hugely appreciated. Thanks!

**Comment by ivanm376** [comment]
temporary solution could be downloading needed archive by hand from https://dl.nwjs.io/ and unzipping it into *./cache* folder, so builder will ignore downloading

**Comment by adam-lynch** [comment]
So this is an error when you're running the app. I'm not sure I understand how skipping the  download helps. Could someone explain? @Varunram which OS are you building on? Does it happen with a simple Hello World app?

**Comment by mohsenno1** [comment]
In my case I can not download the file. It always stops at 66% !!!

**Comment by kessler** [comment]
@Varunram is this still relevant?

**Comment by Varunram** [comment]
@kessler I haven't tried this out in a long while, I guess someone else can open a new issue with updated stats /  this can be open and people can comment on it. Please feel free to close this issue, if needed, thanks!

**Comment by kessler** [comment]
@Varunram thanks for the response! I'll close this for now.

**Comment by YLivay** [comment]
I'm also getting this, but it seems the issue is on dl.nwjs.io's end and not nw-builder, so the suggestion for specifying a mirror in issue #364 sounds like a pretty good solution :)

p.s. was trying to fetch https://dl.nwjs.io/v0.32.1/nwjs-v0.32.1-win-ia32.zip with curl. it errors out with:
```
curl: (18) transfer closed with 37739634 bytes remaining to read
```
The response headers also contained `X-Cache-Status: MISS` despite multiple attempts to download the file.

Trying to fetch it from https://cnpmjs.org/mirrors/nwjs/v0.32.1/nwjs-v0.32.1-win-x64.zip works however.

**Comment by kessler** [comment]
@YLivay Thanks for the info, you're right #364 looks relevant indeed
